packages/pages/src/components/address/methods.ts

```diff
@@ -95,13 +95,15 @@ export const getDirections = (
     }
     default: {
       const listingsMap: { [key in ListingPublisher]?: string } | undefined =
-        listings?.reduce(
-          (obj, listing) => ({
-            ...obj,
-            [listing.publisher.toUpperCase()]: listing.listingUrl,
-          }),
-          {}
-        );
+        listings
+          ?.filter((listing) => listing && listing.publisher != null)
+          ?.reduce(
+            (obj, listing) => ({
+              ...obj,
+              [(listing.publisher || "").toUpperCase()]: listing.listingUrl,
```

Comment:
Since we have the `filter` above, do we need the `|| ""` here? I'm still not sure we actually need to construct a full `listingsMap` either. It seems like `listingsMap` is only used on lines 107 and 108. That is, we just check if GMB is in `listingsMap` and use it if it is. 

Instead of creating `listingsMap`, could you just do:
```
const gmbListing = listings.find(listing => 
    listing && listing.publisher === listingsMap[ListingPublisherOption.GOOGLEMYBUSINESS);

gmbListing && return gmbListing;
...
[Rest of default case]
```

Reply:
I believe we have the `listingsMap` in place for if/when we're going to add more listing publisher options (right now it only checks for Google, but that may change down the line)